### ISSUE TITLE
Unique key updating entity

### ISSUE THREAD

**Issue by manelj** [spotorm/spot2#238]
I think that if i update an entity changing the value of an "unique" field, save system does not check unique integrity.

**Comment by nebulousGirl** [comment]
You are right. The validation of the unique constraint is limited to new entities: https://github.com/spotorm/spot2/blob/master/lib/Mapper.php#L1125

I don't know if there is a reason being that, but the code should still be failing at the database level with an exception.
